### PR TITLE
nit: Use nil pointer pattern to check for interface compliance.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

Reference: https://github.com/cosmos/ibc-go/pull/3437#issuecomment-1504975870

To use a consistent style between creating an empty struct and taking its reference vs. casting a nil pointer to the type.

For types implementing interfaces using value receivers this works in exactly the same way since a pointer to that type is also considered as implementing the interface.

I'm hopeful I got most of them.

closes: #XXXX

### Commit Message / Changelog Entry

```bash
nit: Use nil pointer pattern to check for interface compliance.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/27-interchain-accounts/controller/keeper/grpc_query.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 	icatypes "github.com/cosmos/ibc-go/v7/modules/apps/27-interchain-accounts/types"
 )
 
-var _ types.QueryServer = Keeper{}
+var _ types.QueryServer = (*Keeper)(nil)
```
Comment (DimitrisJim) [inline modules/apps/27-interchain-accounts/controller/keeper/grpc_query.go:14]: I'd be happy to revert the changes on types implementing interfaces using value receivers if it generally gives the impression (unsure) they must be used as pointers.

**Review by chatton — APPROVED** [review]
I think the `nil` version is more efficient, but the gains here would be so inconsequential that I think the decision made should be one that favours readability over performance.

I personally find the struct literal syntax more readable/intuitive, however I don't feel that strongly about it. Consistency is more important to me than the actual style we choose.

Happy to go forward with this style if that is the preference people have!

File: modules/core/04-channel/types/msgs.go [hunk 2]
```diff
@@ -12,7 +12,17 @@ import (
...
+var (
+	_ sdk.Msg = (*MsgChannelOpenInit)(nil)
+	_ sdk.Msg = (*MsgChannelOpenTry)(nil)
+	_ sdk.Msg = (*MsgChannelOpenAck)(nil)
+	_ sdk.Msg = (*MsgChannelOpenConfirm)(nil)
+	_ sdk.Msg = (*MsgChannelCloseInit)(nil)
+	_ sdk.Msg = (*MsgChannelCloseConfirm)(nil)
+	_ sdk.Msg = (*MsgRecvPacket)(nil)
+	_ sdk.Msg = (*MsgAcknowledgement)(nil)
+	_ sdk.Msg = (*MsgTimeout)(nil)
```
Comment (crodriguezvega) [inline modules/core/04-channel/types/msgs.go:24]: Nice that you grouped all these together. :)

**Review by crodriguezvega — APPROVED** [review]
Nice work, @DimitrisJim. :)
